langchain/document_loaders/generic.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Iterator, List, Optional, Sequence, Union
+
+from langchain.document_loaders.base import BaseBlobParser, BaseLoader
+from langchain.document_loaders.blob_loaders import BlobLoader, FileSystemBlobLoader
+from langchain.document_loaders.parsers.registry import get_parser
+from langchain.schema import Document
+from langchain.text_splitter import TextSplitter
+
+_PathLike = Union[str, Path]
+
+
+class GenericLoader(BaseLoader):
```

Comment:
ooc are you imagining for everything that has both a blob parser/parser and doc loader atm we'll reimplement the doc loader as child of this?

Reply:
Mostly thinking of this as living independently as a self contained unit. It may need 2-3 more classmethods and it should be able to load content from most popular locations and then parse is with any arbitrary parser. 

I think this will replace the functionality of a bunch of other loaders, we don't necessarily need to refactor any of the existing ones. Can keep them as they are.